### DISCUSSION TITLE
Build custom longhorn docker images

### DISCUSSION THREAD

**Discussion by vr2388** [longhorn/longhorn#8380] [General]
## Question

How to build custom longhorn docker images.

## Current behavior

We are trying to build longhorn docker images using Dockerfile available in the longhorn github project. But when we build the image, its size is around 2.42GB which is more than the official longhorn images.

I am trying to build longhorn-engine image
github source code: https://github.com/longhorn/longhorn-engine
Expected image size: 123.59 MB
Current image size: 2.42GB
<img width="639" alt="image" src="https://github.com/longhorn/longhorn/assets/133762455/9ab6cf7f-6e2c-47db-a81e-ae3f6c5dfed0">

## Expectation

We need a procedure and commands to build the longhorn images from the source code.

**Comment by mantissahz** [comment]
For `longhorn-engine`, I think there is only one command as this https://github.com/longhorn/longhorn-engine?tab=readme-ov-file#building-from-source-code

```bash
make
```

You must have installed the docker for sure.
You can edit `Dockerfile.dapper` for the building environment, and `package/DockerFile` for the final docker image.
You can edit `script/ci` to skip some steps but steps "build", "validate" and "package" are necessary.
I believe that the `longhorn-engine` image with size 2.42GB is built for the building environment, and it is not the final image.
